pkg/config/expand.go

```diff
@@ -82,6 +86,32 @@ func (dc *DeploymentConfig) addSettingsToModules() {
 	}
 }
 
+func (dc *DeploymentConfig) addMetadataToModules() error {
+	for iGrp, grp := range dc.Config.DeploymentGroups {
+		for iMod, mod := range grp.Modules {
+			if mod.RequiredApis == nil {
+				_, projectIDExists := dc.Config.Vars["project_id"].(string)
+				if !projectIDExists {
```

Comment:
I definitely prefer to live in a world where I know that

- project ID exists and is at least potentially a valid Google Cloud project ID (length, characters, etc.)
- deployment name exists and is at least potentially a valid Google Cloud resource ID prefix (length, characters, etc.)

Reply:
I see, the behavior here isn't any different since the error isn't fatal. How we handle the project_id requirement should be looked at further, and probably kept in line with how we handle deployment name now. 

I'll resolve this and add a task for it.

Reply:
Actual ran a quick test, and realized that we are now warning the user twice if a project doesn't exist:
```
./ghpc create examples/hpc-cluster-small.yaml 
2022/09/07 23:05:37 could not add module metadata: global variable project_id must be defined
deployment variable project_id was not set
```

If this function is called each time `ghpc create` is called, I'd suggest removing the other warning, otherwise I'd suggest dropping the Println for this error and let the other validation catch it.

Reply:
I actually ran into this when deploying networks in two separate projects, I had to fill project_id with a dummy value that was never used. I agree that it probably shouldn't be a requirement. 

Thanks for making the change!